eclass/distutils-r1.eclass

```diff
@@ -2041,6 +2069,23 @@ _distutils-r1_post_python_install() {
 			eerror "https://projects.gentoo.org/python/guide/qawarn.html#stray-top-level-files-in-site-packages"
 			die "Failing install because of stray top-level files in site-packages"
 		fi
+
+		if [[ ! ${DISTUTILS_EXT} && ! ${_DISTUTILS_EXT_WARNED} ]]; then
+			local suffix=$(
+				"${EPYTHON}" - <<-EOF || die
+					import sysconfig
+					print(sysconfig.get_config_var("EXT_SUFFIX"))
+				EOF
+			)
+
+			if [[ $(find "${sitedir}" -name "*${suffix}" | head -n 1) ]]
```

Comment:
Should this check for `*.abi3.so` too? Not super familiar with these really, but python loads these and dev-python/adblock toss a adblock.abi3.so (rust) in site-packages.

Reply:
No clue.

Reply:
Heh, apparently it seems that Python is still happy to even load plain `sip.so`, sigh. FWICS, `.abi3.so` is quite common.

OTOH, no special suffix seems to be used by 2 packages: `dev-python/sip` and `dev-python/dbus-python`. There's also irrelevant file installed by pydevd with no special suffix but pydevd installs C extensions anyway, so it wouldn't trigger a false positive.

I suppose the simplest solution would be to find all .so files after all.